db/db_utils.go

```diff
@@ -525,15 +525,18 @@ func AggregateContext(ctx context.Context, collection string, pipeline interface
 // AggregateWithMaxTime runs aggregate and specifies a max query time which
 // ensures the query won't go on indefinitely when the request is cancelled.
 func AggregateWithMaxTime(collection string, pipeline interface{}, out interface{}, maxTime time.Duration) error {
-	session, database, err := GetGlobalSessionFactory().GetSession()
+	ctx, cancel := context.WithTimeout(context.Background(), maxTime)
+	defer cancel()
+
+	session, database, err := GetGlobalSessionFactory().GetContextSession(ctx)
 	if err != nil {
 		err = errors.Wrap(err, "establishing DB connection")
 		grip.Error(err)
 		return err
 	}
 	defer session.Close()
 
-	return database.C(collection).Pipe(pipeline).MaxTime(maxTime).All(out)
+	return database.C(collection).Pipe(pipeline).All(out)
```

Comment:
Timeouts with MaxTime is removed from our Anser repository and instead, the context passed in should use a timeout. This is a breaking change from v2, the go driver no longer exposes any methods to control max time / timeouts.